res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -1571,6 +1572,8 @@ PioneerDDJSX.panelSelectButton = function(channel, control, value, status, group
 
 PioneerDDJSX.shiftPanelSelectButton = function(channel, control, value, status, group) {
     var channelGroup;
+    PioneerDDJSX.shiftPanelSelectPressed = value ? true : false;
```

Comment:
You understood that correctly, but it's very easy to hold shift and panel select with one hand. In either way you need two hands to assign EffectUnit 3 and 4. I can put my thumb onto the shift button and my forefinger onto the panel select easily while pressing the assignment buttons with the other hand. I have chosen this way because then the original panel select function can be maintained and I think it's more intuitive to activate/deactivate panels (samplers, effects) if you don't press shift (as the button is only labeled PANEL SELECT).